### ISSUE TITLE
Support sqlite

### ISSUE THREAD

**Issue by simonschaufi** [Nimut/TYPO3-testing-framework#131]
Since TYPO3 officially supports sqlite databases, this extension should as well.

See https://github.com/TYPO3/testing-framework/pull/79 which implemented this in TYPO3 testing-framework and maybe also https://github.com/TYPO3/testing-framework/pull/121

```bash
typo3DatabaseDriver=pdo_sqlite .Build/bin/phpunit -c .Build/vendor/nimut/testing-framework/res/Configuration/FunctionalTests.xml Tests/Functional/
```

This command fails because of `setDBinit` configuration. It actually crashes in the `initializeTestDatabase` method.

When I set the value to an empty string, I still get an error if 2 following tests are executed with the second iteration in the method `setUpTestDatabase` and the `$schemaManager->listDatabases()` call. This is not supported in sqlite:

```
Doctrine\DBAL\DBALException: Operation 'Doctrine\DBAL\Platforms\AbstractPlatform::getListDatabasesSQL' is not supported by platform.
```

**Comment by IchHabRecht** [comment]
Hi @simonschaufi,

are your tests running with the typo3/testing-framework?

**Comment by simonschaufi** [comment]
I haven't tried it but I just wanted to ask you anyway why there are 2 frameworks for the same purpose. Should I switch to typo3/testing-framework or what is the advantage of THIS framework?

**Comment by IchHabRecht** [comment]
This testing-framework is able to be used for supporting multiple TYPO3 versions as extensions normally do/want. As far as I know the `setUpTestDatabase` code should be the same for both testing frameworks. That's why I asked if you ever tried to run your tests on sqlite with the typo3/testing-framework.

**Comment by simonschaufi** [comment]
Yes, when I exchange the testing framework then the tests run fine with sqlite.

**Comment by IchHabRecht** [comment]
Hi @simonschaufi,

Would you mind to test the pull request #132 to see if this fixes your tests?

**Comment by simonschaufi** [comment]
Sure, what about this PR https://github.com/TYPO3/testing-framework/pull/121?

**Comment by IchHabRecht** [comment]
I'm currently not able to spend more time in this support. If this is important for you, please stay with the typo3/testing-framework. I just tried to fix your recommendation to be able to run tests with SQLite

**Comment by simonschaufi** [comment]
ok. I will give feedback in the evening

**Comment by simonschaufi** [comment]
In method `getDatabaseConfiguration` the line should look like this:

`if ($databaseName || $databaseHost || $databaseUsername || $databasePassword || $databasePort || $databaseSocket || $databaseDriver) {`

then it works. Thank you for the quick pr!

**Comment by simonschaufi** [comment]
@IchHabRecht Thank you very much! :orange_heart:

**Comment by IchHabRecht** [comment]
Hi @simonschaufi,

I just released a new version 5.0.4 with the fix. Thank you for your help and patience.
